platform/include/platform/mbed_rtc_time.h

```diff
@@ -32,8 +32,10 @@ extern "C" {
  * @{
  */
 
-/* Timeval definition for non GCC_ARM toolchains */
-#if !defined(__GNUC__) || defined(__clang__)
+/* Timeval definition for non GCC_ARM toolchains,
+ * note: _TIMEVAL_DEFINED is set e.g. in arm-none-eabi/include/sys/_timeval.h
+ */
+#if !defined(__GNUC__) && !defined(_TIMEVAL_DEFINED)
```

Comment:
Why are we removing `__clang__` ?

Reply:
Because it was causing the double-defined errors when using clang-tidy.
AFAIK clang is not supported as a compiler for Mbed anyway, so it shouldn't break anything, but this change allows usage of clang-tidy on a compilation database generated for the GCC toolchain.

I've tried to check for defines used in the headers declaring the struct, but there is no consistency between them, so the alternative solution would be to cover all the known define flags and decide whether do define `struct timeval` based on that.
The defines I've found so far are:
* The `libc` and derivatives (e.g. Ubuntu/Debian package `libc6-dev`) provides `#define __timeval_defined 1`, e.g. in `/usr/include/<architecture_specific>/bits/types/struct_timeval.h`
* The `newlib` (e.g. package `libnewlib-dev`) provides `#define _TIMEVAL_DEFINED`, e.g. in `/usr/include/newlib/sys/_timeval.h`
I don't have ARM's `microlib` at hand to inspect what defines it does provide.

Reply:
if we define our own, and check if not yet defined (`_TIMEVAL_DEFINED`), we should define it.
adding on line 39 `#define _TIMEVAL_DEFINED` so anyone getting this header, can check if its defined and it is (by us)

Reply:
we define gnu flag for compatibility for ARMClang, so `__GNUC__` is defined. I assume ARMClang do not have defined `_TIMEVAL_DEFINED` (seeing similar implementation https://github.com/cesanta/mjs/blob/master/mjs.c#L1054 for mbed). 

I'll fetch this to test with ARMClang, I cant quickly find in the online docs.

Reply:
Maybe the most compatible solution would be to drop the `defined(__GNUC__)` check whatsoever and decide only based on the existance of the `_TIMEVAL_DEFINED` (and its variants)?

Reply:
I think so then. I'll check if we use in tests timeval so our CI tests this with both  toolchains we support.

Reply:
I've pushed a version with checking for the defines - let me know if there are any more conditions we should check.